framework/src/utils/SlepcSupport.C

```diff
@@ -36,7 +36,7 @@ InputParameters
 getSlepcValidParams(InputParameters & params)
 {
   MooseEnum solve_type("POWER ARNOLDI KRYLOVSCHUR JACOBI_DAVIDSON "
-                       "NONLINEAR_POWER NEWTON");
+                       "NONLINEAR_POWER NEWTON PJFNK JFNK");
```

Comment:
Should we default this to PJFNK as the Steady?

Reply:
Oh, one more thing: because we have this now, we possibly do not need this parameter `matrix_free`.

Reply:
We need `matrix_free` for others than Newton.

Reply:
> Should we default this to PJFNK as the Steady?

Yes, let me try if we could make that happen.

Reply:
Then let us add a document for the `matrix_free` parameter, something like that it is works with this list of solve types... for the other list of solve types, it is inactive. Something like this. We cannot assume users will use the combination properly unless we made them aware.

Reply:
Or maybe add `JF_POWER JF_ARNOLDI ...` for all the others. I feel your original design is good. But since we need to add JFNK and PJFNK and reinterprete NEWTON, we can add the JF version of the other types in the MOOSE ENUM.

Reply:
"JF" does not make sense for other algorithms. It was why I had `matrix_free` not `Jacobian_free`.

Then I have to double the number of algorithms: `MF_A, A, MF_B , B, MF_C, C, MF_D, D`.  @loganharbour Do you like this?

Reply:
How the other algorithms do matrix-free calculations? Are you specifically referring Jacobian free as for the matrix-vector product, which the other algorithms do not have? To be clear, I like your original design. But since we need to make Eigenvalue more consistent with Steady. We possible need to do this. Another possiblility is to rename `matrix_free` to `matrix_free_for_non_newton_methods`. But I guess you do not like it.

Reply:
I think the right solution is: we should error out if users try to specify matrix_free parameter for Newton /PJFNK/JFNK. 

"Please do not specify matrix_free for  xxx"

Reply:
You are right, I hate a long name `matrix_free_for_non_newton_methods`

Reply:
> I think the right solution is: we should error out if users try to specify matrix_free parameter for Newton /PJFNK/JFNK.

This involves API upgrade, and we can not do it in this PR.

Reply:
Error sounds good and we can do it later. Meanwhile more documentation is always good.